lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,173 @@
+            logger.error("Failed to create temporary directory")
             sys.exit(1)
 
-        runs_copied += 1
+        user = self.context.user if self.context.user else "pbench"
+        prefix = self.context.prefix if self.context.prefix else ""
+
+        runs_copied = 0
+        failures = 0
 
-    if runs_copied + failures > 0:
-        logger.debug(
-            "successfully moved %s runs, encountered %s failures", runs_copied, failures
-        )
+        for dirent in config.pbench_run.iterdir():
+            if not self.dirent.is_dir():
+                continue
+            if dirent.name.startswith("tools-") or dirent.name == "tmp":
+                continue
+            result_dir = dirent
+            mrt = MakeResultTb(result_dir, temp_dir, user, prefix, self.config, logger)
+            make_return_val = mrt.check_result_target_dir(result_dir, temp_dir)
+            if make_return_val:
+                sys.exit(1)
+            status, result_tb_name = mrt.make_result_tb()
+
+            if status == 0:
+                if os.path.exists(result_tb_name):
+                    assert (
+                        result_tb_name
+                    ), "Logic bomb!  make_result_tb() always returns a tar ball name"
+                    crt = CopyResultTb(controller, result_tb_name, self.config, logger)
+                    copy_return_val = crt.check_tb_tbmd5()
+                    if copy_return_val:
+                        sys.exit(1)
+                    ret = crt.copy_result_tb(self.context.token)
+                    if ret:
+                        sys.exit(1)
+            else:
+                logger.error("Error occured while creating Result Tarball")
+                failures += 1
+                continue
+
+            try:
+                # We always remove the constructed tar ball, regardless of success
+                # or failure, since we keep the result directory below on failure.
+                os.remove(result_tb_name)
+                os.remove(f"{result_tb_name}.md5")
+            except OSError:
+                logger.error("rm failed to remove %s and its .md5 file", result_tb_name)
+                failures += 1
+                continue
+
+            try:
+                shutil.rmtree(result_dir)
+            except OSError:
+                logger.error(
+                    "rm failed to remove the %s directory hierarchy", result_dir
+                )
+                failures += 1
+                continue
+
+            runs_copied += 1
+
+        if runs_copied + failures > 0:
+            logger.debug(
+                "successfully moved %s runs, encountered %s failures",
+                runs_copied,
+                failures,
+            )
+            return 1
+
+        return 0
+
+
+@click.command()
+@common_options
+@click.option(
+    "--controller", help="Controller name (should be hostname -f value)",
+)
+@click.option(
+    "--user", help="Pbench user",
```

Comment:
Maybe "Pbench server username"?? Although we do have to decide on exactly how we're going to end up labeling a user: e.g., username or email. Or do we allow either? Hmm...

Reply:
No, it isn't Pbench server username. It is a 'user' argument that was already available in pbench-move-results script which was basically an environment variable {PBENCH_USER}

Reply:
So the consensus (see conversation below) is that we can and should remove this command line option, so I think this conversation can be marked "resolved".

Reply:
Sort of.

Thinking of compatibility, we really should accept the parameters `--user` and `--prefix` and issue a warning saying they are no longer used.  And then in a future release remove them.

Reply:
Click provides nice support for deprecating _commands_, but, sadly, not for command line options.  However, it shouldn't be hard to default their values to `None` and test after argument processing to see if they have any other value and issue a warning if they do.

The important thing is to ensure that the code isn't using these values anymore.

Reply:
Ok I will make sure about this

Reply:
and add another PR to remove them completely

Reply:
I think Peter concluded that we cannot remove them completely until a future release, so there is no push for an additional PR -- we just want to add a warning for people in the current release.

Reply:
yeah ok :)